subsys/net/l2/Kconfig

```diff
@@ -16,11 +16,11 @@ config NET_L2_DUMMY
 config NET_L2_BT
 	bool "Enable Bluetooth support"
 	depends on NET_IPV6
-	select BT
-	select BT_PERIPHERAL
-	select BT_CENTRAL
-	select BT_SMP
-	select BT_L2CAP_DYNAMIC_CHANNEL
+	depends on BT
+	depends on BT_PERIPHERAL
+	depends on BT_CENTRAL
+	depends on BT_SMP
+	depends on BT_L2CAP_DYNAMIC_CHANNEL
 	select NET_6LO
```

Comment:
Don't know exactly why it would go into an infine loop, expect perhaps if there is a circular dependecy somewhere, but it really sounds bad that it apparently on affect BT_ options not NET_ options so application are forced to mark the 5 extra options.

Reply:
It is a circular dependency, starting at BT_CTRL => BT => NET_BUF .... => BT_CTRL again ([kconfig-loop.txt](https://github.com/zephyrproject-rtos/zephyr/files/2918449/kconfig-loop.txt)). And the problem goes deeper than here, really. There are many more places selecting symbols in a way that's prone to these problems. IMO since the implementation of Kconfig (that resolves the symbols) can't resolve dependencies automagically (see issue #8181) we have to force users to be explicit. It is better that Kconfig fails with a message [1]:

> waning: FOO (defined in ..) was assigned the value 'y' but got the value 'n'

than having a huge circular dependency to fix because of symbols selecting each other uncritically. 

[1] This error message could be improved though, it could say
> FOO was assigned 'y' but got the value 'n' because one or more of its dependencies were not met.
> FOO depends on BAR (=n) && BAZ (=y)

Reply:
> It is a circular dependency, starting at BT_CTRL => BT => NET_BUF .... => BT_CTRL again ([kconfig-loop.txt](https://github.com/zephyrproject-rtos/zephyr/files/2918449/kconfig-loop.txt)).

I guess the dependecy is BT_CTRL -> ... -> BT_LL_CHOISE -> BT_CTRL, I don't see how NET_BUF would cause this since it has no dependencies something else is causing this circular depency. In fact I think this is related to BT_CTRL as NET_L2_BT don't have a depency to it (btw it seems you guys are trying to enable openthread so I wonder why NET_L2_BT or you want both 15.4 an BT L2 drivers?)

Reply:
If you have a another/better way of dissolving this circular dependency I am happy to add it. I don't think there is only one here. I simply removed using `select` by net/l2. I am not trying to enable openthread, I just want to resolve the circular dependency.

Do you have a suggestion for what I should do instead?

Reply:
Can you paste the .conf file in use in this case, anyway there got to be a way to fix circular dependencies somehow, perhaps switch to imply or you tried that already?

Reply:
`imply` does not change anything, unfortunately.

[.config](https://github.com/zephyrproject-rtos/zephyr/files/2919499/dot_config.txt)